### PR TITLE
Check we're at the HTTP handler root path

### PR DESCRIPTION
**Pull request by jenting**
The "/" pattern matches everything, so we need to check
that we're at the HTTP handler root path.

If the path does not match the existing registered path, return HTTP
status code 301 moved permanently.

Signed-off-by: JenTing Hsiao <jenting.hsiao@suse.com>

### CODE REVIEW

**Comment by jenting** [comment]
Merge master branch.

**Comment by jenting** [comment]
@stevesloka PTAL if you have time.

File: cmd/gangway/main.go [hunk 1]
```diff
@@ -47,6 +47,18 @@ func httpLogger(fn http.HandlerFunc) http.HandlerFunc {
 	}
 }
 
+func rootPathHandler(fn http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		// The "/" pattern matches everything, so we need to check
+		// that we're at the root here.
+		if cfg.HTTPPath == "" && r.URL.Path != "/" {
+			http.NotFound(w, r)
```
Comment (stevesloka) [inline cmd/gangway/main.go:55]: Maybe this should just 301 redirect back to `/`?

Comment (jenting) [reply cmd/gangway/main.go:55]: Sounds good.

**Pull request closed without merging**